cou/steps/analyze.py

```diff
@@ -160,7 +181,15 @@ def current_cloud_os_release(self) -> Optional[OpenStackRelease]:
         :return: OpenStack release codename
         :rtype: OpenStackRelease
         """
-        return min(
-            (app.current_os_release for app in self.apps_control_plane + self.apps_data_plane),
-            default=None,
+        control_plane = (
+            [self.min_os_version_control_plane] if self.min_os_version_control_plane else []
         )
+        data_plane = [self.min_os_version_data_plane] if self.min_os_version_data_plane else []
+        return min(control_plane + data_plane, default=None)
+
+    def manually_upgrade_data_plane(self) -> None:
+        """Warning message to upgrade data plane charms if necessary."""
+        if self.min_os_version_control_plane and self.min_os_version_data_plane:
+            if self.min_os_version_control_plane > self.min_os_version_data_plane:
+                data_plane_apps = ", ".join([app.name for app in self.apps_data_plane])
+                print(f"WARNING: Please upgrade manually the data plane apps: {data_plane_apps}")
```

Comment:
To my this should not be part of analyze, it's saying upgrade plan so it's should be part of generate plan. What do you say?